core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -67,20 +67,20 @@ public void setDocument(int docId) {
     }
 
     @Override
-    public Object get(Object key) {
+    public ScriptDocValues<?> get(Object key) {
         // assume its a string...
         String fieldName = key.toString();
```

Comment:
This means you could get a non-null value, even for a non-null key, which I find a bit confusing given the generics say that the keys are strings? Maybe we should change it to a `Map<Object, ScriptDocValues<?>>` or stop calling `toString()` and just return `null` if the key is not a `String`?